test/e2e/konnect_test.go

```diff
@@ -68,6 +70,71 @@ func TestKonnectConfigPush(t *testing.T) {
 	requireKonnectNodesConsistentWithK8s(ctx, t, env, rgID, cert, key)
 }
 
+func TestKonnectLicenseActivation(t *testing.T) {
+	t.Parallel()
+	skipIfMissingRequiredKonnectEnvVariables(t)
+
+	ctx, env := setupE2ETest(t)
+
+	rgID := createTestRuntimeGroup(ctx, t)
+	cert, key := createClientCertificate(ctx, t, rgID)
+	createKonnectClientSecretAndConfigMap(ctx, t, env, cert, key, rgID)
+
+	manifestFile := "../../deploy/single/all-in-one-dbless-konnect-enterprise.yaml"
+	t.Logf("deploying %s manifest file", manifestFile)
+
+	manifest := getTestManifest(t, manifestFile)
+	deployKong(ctx, t, env, manifest)
+
+	exposeAdminAPI(ctx, t, env, types.NamespacedName{Namespace: "kong", Name: "proxy-kong"})
+
+	t.Log("disabling license management")
+	kubeconfig := getTemporaryKubeconfig(t, env)
+	cmd := exec.Command("kubectl", "--kubeconfig", kubeconfig, "set", "env", "-n", "kong", "deployment/ingress-kong",
+		"CONTROLLER_KONNECT_LICENSING_ENABLED-")
+	stdout, stderr := new(bytes.Buffer), new(bytes.Buffer)
+	cmd.Stdout = stdout
+	cmd.Stderr = stderr
+	err := cmd.Run()
+	require.NoErrorf(t, err, "disabling licensing failed: STDOUT(%s) STDERR(%s)", stdout.String(), stderr.String())
```

Comment:
How about extracting a `setEnv(t *testing.T, namespace, name, env, value string)` and using it in the test (here and below when setting back to `true`) to make the test body more concise?

Reply:
4b1acbcfaa3efcd58e3ba66019fa0f0a879888cf